// react-native-network-logger/lib/module/components/RequestList.js
import React from 'react';
import { View, StyleSheet, FlatList } from 'react-native';
import { useThemedStyles } from '../theme';
import ResultItem from './ResultItem';
import Button from './Button';

const RequestList = ({
  requests,
  onPressItem,
  onShowMore
}) => {
  const styles = useThemedStyles(themedStyles);
  return /*#__PURE__*/React.createElement(View, {
    style: styles.container
  }, /*#__PURE__*/React.createElement(FlatList, {
    keyExtractor: (item, index) => index.toString(),
    ListHeaderComponent: () => /*#__PURE__*/React.createElement(Button, {
      onPress: onShowMore,
      style: styles.more
    }, "More"),
    data: requests,
    renderItem: ({
      item
    }) => /*#__PURE__*/React.createElement(ResultItem, {
      request: item,
      onPress: () => onPressItem(item)
    })
  }));
};

const themedStyles = theme => StyleSheet.create({
  container: {
    backgroundColor: theme.colors.background,
    flex: 1
  },
  more: {
    marginLeft: 10
  }
});

export default RequestList;
