// react-native-network-logger/lib/commonjs/components/RequestList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _theme = require("../theme");

var _ResultItem = _interopRequireDefault(require("./ResultItem"));

var _Button = _interopRequireDefault(require("./Button"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const RequestList = ({
  requests,
  onPressItem,
  onShowMore
}) => {
  const styles = (0, _theme.useThemedStyles)(themedStyles);
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.container
  }, /*#__PURE__*/_react.default.createElement(_reactNative.FlatList, {
    keyExtractor: (item, index) => index.toString(),
    ListHeaderComponent: () => /*#__PURE__*/_react.default.createElement(_Button.default, {
      onPress: onShowMore,
      style: styles.more
    }, "More"),
    data: requests,
    renderItem: ({
      item
    }) => /*#__PURE__*/_react.default.createElement(_ResultItem.default, {
      request: item,
      onPress: () => onPressItem(item)
    })
  }));
};

const themedStyles = theme => _reactNative.StyleSheet.create({
  container: {
    backgroundColor: theme.colors.background,
    flex: 1
  },
  more: {
    marginLeft: 10
  }
});

var _default = RequestList;
exports.default = _default;
